apps/prairielearn/elements/pl-rich-text-editor/pl-rich-text-editor.py

```diff
@@ -100,6 +100,38 @@ def render(element_html: str, data: pl.QuestionData) -> str:
     )
     element_text = element_inner_html(element)
 
+    submitted_files = data["submitted_answers"].get("_files", [])
+    submitted_file = next(
+        (f for f in submitted_files if f.get("name", None) == file_name), None
+    )
+
+    if data["ai_grading"]:
+        if data["panel"] != "submission" or not submitted_file:
+            return ""
+
+        contents = submitted_file.get("contents", "")
+        contents = base64.b64decode(contents).decode("utf-8")
+
+        # MathJax gets embedded as `span.ql-formula` elements. This is fine;
+        # we'll just remove all attributes to save on tokens.
+        html_doc = lxml.html.fragments_fromstring(contents)
+        for fragment in html_doc:
+            if isinstance(fragment, str):
+                continue
+
+            for span in fragment.xpath('.//span[@class="ql-formula"]'):
+                span.attrib.clear()
+
+        # Reconstruct the HTML content
+        contents = "".join([
```

Comment:
```suggestion
        contents = "".join(
```
Does not need to be put into a list, can be done with the generator expression directly.

Reply:
Done in 9e5e9c7.